src/coreclr/build.sh

```diff
@@ -38,45 +33,23 @@ usage_list+=("-skipnuget: skip NuGet package generation.")
 usage_list+=("-skiprestore: specify the official build ID to be used by this build.")
 usage_list+=("-skiprestoreoptdata: build CoreLib as PartialNGen.")
 usage_list+=("-staticanalyzer: skip native image generation.")
-usage_list+=("-stripSymbols: skip native image generation.")
 
-setup_dirs()
+setup_dirs_local()
 {
-    echo Setting up directories for build
+    setup_dirs
 
-    mkdir -p "$__RootBinDir"
-    mkdir -p "$__BinDir"
     mkdir -p "$__LogsDir"
     mkdir -p "$__MsbuildDebugLogsDir"
-    mkdir -p "$__IntermediatesDir"
 
-    if [ $__CrossBuild == 1 ]; then
+    if [ "$__CrossBuild" = 1 ]; then
         mkdir -p "$__CrossComponentBinDir"
     fi
 }
 
-# Check the system to ensure the right prereqs are in place
-
-check_prereqs()
-{
-    echo "Checking prerequisites..."
-
-    # Check presence of CMake on the path
-    hash cmake 2>/dev/null || { echo >&2 "Please install cmake before running this script"; exit 1; }
-
-    function version { echo "$@" | awk -F. '{ printf("%d%02d%02d\n", $1,$2,$3); }'; }
-
-    local cmake_version=$(cmake --version | grep -Eo "[0-9]+\.[0-9]+\.[0-9]+")
-
-    if [[ $(version $cmake_version) -lt $(version 3.14.0) ]]; then
-        echo "Please install CMake 3.14 or newer from http://www.cmake.org/download/ or https://apt.kitware.com and ensure it is on your path."; exit 1;
-    fi
-}
-
 restore_optdata()
 {
     local OptDataProjectFilePath="$__ProjectRoot/src/.nuget/optdata/optdata.csproj"
-    if [[ ( $__SkipRestoreOptData == 0 ) && ( $__isMSBuildOnNETCoreSupported == 1 ) ]]; then
+    if [ "$__SkipRestoreOptData" = 0 ] && [ "$__IsMSBuildOnNETCoreSupported" = 1 ]; then
```

Comment:
single `if [ expr ]` is a short form of `if test expr` which is posix shell complaint and uses single `=` for equality check. we are inconsistently using `if [ x = y]`, `if [[ x == y ]]`, `if [ x == y ]` in various places. i just picked up the posix shell flavor and used it everywhere.

Reply:
Could we still use the == instead of = or could it break some shells? I was trying to find a list of comparison operators for posix shells and while I have found some documents on the syntax including grammar, the operators were not there. Do you have a pointer to a document that I can read on that?

Reply:
* test(1): https://linux.die.net/man/1/test
* [The Unix Shell's Humble If](https://thoughtbot.com/blog/the-unix-shells-humble-if)

these indicate that `=` is used for string comparison and `-eq` is used for numeric comparison. We can also use `(( x == 1 ))` for numeric comparison.

However, second link also mentions:

```
 Best Practice: If you have no reason to target POSIX shell, use [[
Bracket-Bracket

[[ comes with the following features over the normal test command:

*    Use familiar ==, >=, and <= operators
```

which is the case here: `#!/usr/bin/env bash`.

Reply:
Thank you for the pointers. I would then prefer to unify on the `[[` instead of `[`.